src/OrchardCore/OrchardCore.DisplayManagement/ShapeFactoryExtensions.cs

```diff
@@ -0,0 +1,187 @@
+using System.Collections.Concurrent;
+using Castle.DynamicProxy;
+using OrchardCore.DisplayManagement.Shapes;
+using OrchardCore.DisplayManagement.Views;
+
+namespace OrchardCore.DisplayManagement;
+
+public static class ShapeFactoryExtensions
+{
+    private static readonly ConcurrentDictionary<Type, Type> _proxyTypesCache = [];
+    private static readonly ProxyGenerator _proxyGenerator = new();
+    private static readonly Func<ValueTask<IShape>> _newShape = () => ValueTask.FromResult<IShape>(new Shape());
+
+    /// <summary>
+    /// Creates a new generic shape by copying the properties of an object.
+    /// </summary>
+    /// <param name="factory">The <see cref="IShapeFactory"/>.</param>
+    /// <param name="shapeType">The type of shape to create.</param>
+    /// <param name="model">The model to copy.</param>
+    /// <returns></returns>
+    public static ValueTask<IShape> CreateAsync<TModel>(this IShapeFactory factory, string shapeType, TModel model)
+        => factory.CreateAsync(shapeType, Arguments.From(model));
```

Comment:
```suggestion
        => CreateAsync(shapeType, Arguments.From(model));
```

Reply:
How is this going to even build? These are extensions not standard static methods.

Reply:
It could work if it's `CreateAsync(factory, shapeType, Arguments.From(model));`. Anyhow leave these changes and address this comment https://github.com/OrchardCMS/OrchardCore/pull/16715#discussion_r1760340559